### ISSUE TITLE
Show amount of total referrals

### ISSUE THREAD

**Issue by adriaanvanrossum** [simpleanalytics/roadmap#39]
We got this request from @marcelhagedoorn

> I'm tracking my monthly referrals (to track growth) but i have to add up and count all referrals.

![CleanShot 2019-11-12 at 17 01 35@2x](https://user-images.githubusercontent.com/1079135/68662026-94d8d080-056e-11ea-8490-e9bcb5aa397d.jpg)

**Comment by amritnagi** [comment]
I think it would look cleaner if you just put it at the bottom ?

![image](https://user-images.githubusercontent.com/25029371/68663951-68559200-0537-11ea-88e8-543d93ccb861.png)

**Comment by amritnagi** [comment]
Although thinking about it, should direct be included as a referral?

**Comment by adriaanvanrossum** [comment]
Nice, thanks!

> Although thinking about it, should direct be included as a referral?

Hm, it's not really a referral, right? It's more like a source of traffic (direct, referrals)

**Comment by amritnagi** [comment]
Yeah that's true.

How about going super clean:

![image](https://user-images.githubusercontent.com/25029371/68666138-36dec580-053b-11ea-96bb-67b046e3824f.png)

**Comment by adriaanvanrossum** [comment]
That is cool! Let me think about this. Only downside is that it also looks like a loader a bit.

**Comment by amritnagi** [comment]
Regular pie chart seemed a bit out of sync with the rest of the design!

![image](https://user-images.githubusercontent.com/25029371/68666592-1a8f5880-053c-11ea-87f9-556fca4e1d61.png)

**Comment by khrome83** [comment]
I like the donut chart, but I kind of like it is at the bottom and with the numbers always showing. If it was thicker, it would not appear as much like a loader... But that is a valid concern.

**Comment by amritnagi** [comment]
If the donut chart is preferred - perhaps change the loading animation??

I like the three dots personally (last one on 2nd row)
https://github.com/loadingio/css-spinner/

**Comment by adriaanvanrossum** [comment]
Yeah, if we change the loading indicator it's better, I agree.

**Comment by khrome83** [comment]
Out of curiosity, since you know referrals, can you do a breakdown like this (in a donut chart maybe?)

1993 visits Referrals (25% total)

37% Social (Known social networks)
46% SEO (Known search engines)
NN% Direct (no referrer)

OR something like that? it would offer some richer insights and may not be that hard if you had a small dictionary of referrers to flag as Social or SEO.

**Comment by adriaanvanrossum** [comment]
Hm, maybe we could use a toggle like this:

- 150 search
- 120 social
- 60 news.ycombinator.com

and when you click on search you get:

- 150 search
  - 130 google.com
  - 20 duckduckgo.com
- 120 social
- 60 news.ycombinator.com

with some kind of nice animation or something

**Comment by amritnagi** [comment]
An accordion might be perfect for this!

**Comment by khrome83** [comment]
Yeah, I don't think I hate that. 

You may want to do an expander (no auto-closing the previously opened), just because some people may want to see all stats. 

But I love the idea of grouping it. It makes sense to me. 

This may be a bad idea, but since you are so "open". You may want to make the "seed" files to define social/search engines live in GitHub. Then people could do pull requests to update or add to them. 

This would let you add other categories later as it made sense. Though the traditional search/social/direct/referral/email may be right. 

Ohh email is another one that would be pretty easy. A lot of newsletters use substack or revue or something else that is identifiable.

**Comment by amritnagi** [comment]
Maybe something like this? https://codepen.io/abergin/pen/ihlDf

**Comment by khrome83** [comment]
pretty nice...

I would think a right-facing arrow that faces down when expanded might make more sense. The animation is a little glitchy/fast also.

**Comment by amritnagi** [comment]
Do you think it would be good to have grouping on the pages which uses a subfolder hierachy too?

![image](https://user-images.githubusercontent.com/25029371/70708007-6c5d0700-1cd1-11ea-8d86-730c3e4a8884.png)


Also wondering if it should be an option people can turn on and off in their settings?

**Comment by khrome83** [comment]
I would put the arrow on the front. The text below makes it look bad, but it may work better in a design?

What would you do for long URL schemas?

v (100) /components/
  v (20) /components/base/
     (5)     /components/base/button/
      > (10) /components/base/avatar/
     (5)     /components/base/textfield/
  (80) /components/marketing/

Just curious if this is helpful for people?

**Comment by amritnagi** [comment]
Yeah, i thought they would be better at the front too, but it was just a quick mockup.

The long urls get truncated at present, so I'd expect the same.
I would probably drop the subfolder for the drop downs...

![image](https://user-images.githubusercontent.com/25029371/70708981-b8a94680-1cd3-11ea-9e6b-6a42a6fc54e2.png)

This is how i would do it:
If grouped, have 3 dots after the path... and then hovering over it would give you the necessary marker
![image](https://user-images.githubusercontent.com/25029371/70709470-dd51ee00-1cd4-11ea-9b4a-c3843aff3655.png)

I'm 50-50 on it.  I prefer it, as UI wise it looks way cleaner.  
But it depends on how users want to interpret their stats.  In my case, i have all the templates under the templates folder, so it would be nice for me to view them in order under there.
(Other users may have their blog posts etc)

Some users may just want to see which pages are popular overall - perhaps those sites which do not have a lot of pages?

Hence suggesting if it would be useful as an option.

Also: another accordion option - a vue version , might be easier for Adriaan to include this
https://codepen.io/dimaZubkov/pen/MzazgW
